File: app/src/main/java/com/dwf/dontwastefood/MainActivity.java
package com.dwf.dontwastefood;

import android.content.res.AssetManager;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.widget.TextView;

import com.dwf.dontwastefood.proto.UserModel;
import com.dwf.dontwastefood.proto.UserModel.User;
import com.google.protobuf.Message;
import com.google.protobuf.TextFormat;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    public static List<User> userList;

    private static final String TAG = "MainActivity";

    private TextView mTextMessage;
    private AssetManager mAssetManager;
    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    mTextMessage.setText(R.string.title_home);
                    return true;
                case R.id.navigation_recipe:
                    mTextMessage.setText(R.string.title_recipe);
                    return true;
                case R.id.navigation_profile:
                    mTextMessage.setText(R.string.title_profile);
                    return true;
                case R.id.navigation_favorite:
                    mTextMessage.setText(R.string.title_favorite);
                    return true;
            }
            return false;
        }
    };

    private void loadData() throws IOException {
        UserModel.UsersOrBuilder users = UserModel.Users.newBuilder();
        InputStream input = mAssetManager.open("users.prototext");
        InputStreamReader reader = new InputStreamReader(input);
        TextFormat.Parser.newBuilder().build().merge(reader, (Message.Builder) users);
        userList = users.getUserList();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mAssetManager = getAssets();
        mTextMessage = findViewById(R.id.message);
        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
    }

}

File: app/src/main/java/com/dwf/dontwastefood/MainActivity.kt
package com.dwf.dontwastefood

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentTransaction
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.dwf.dontwastefood.proto.UserModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(),
        RecipeCategoryFragment.OnListFragmentInteractionListener,
        HomeFragment.OnFragmentInteractionListener {


    private val fragmentManager: FragmentManager = supportFragmentManager

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        val transaction = fragmentManager.beginTransaction()
        when (item.itemId) {
            R.id.navigation_home -> {
                transaction
                        .replace(fragment_container.id, HomeFragment.newInstance("", ""))
                        .setTransition(FragmentTransaction.TRANSIT_ENTER_MASK)
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_recipe -> {
                transaction
                        .replace(fragment_container.id, RecipeCategoryFragment.newInstance())
                        .setTransition(FragmentTransaction.TRANSIT_ENTER_MASK)
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_member_centre -> {
                transaction
                        .replace(fragment_container.id, CustomerMemberCentreFragment.newInstance("", ""))
                        .setTransition(FragmentTransaction.TRANSIT_ENTER_MASK)
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_favorite -> {
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)

        mUser = intent.getSerializableExtra(LoginActivity.EXTRA_USER) as UserModel.User?

        fragmentManager.beginTransaction()
                .replace(fragment_container.id, HomeFragment.newInstance("", ""))
                .setTransition(FragmentTransaction.TRANSIT_ENTER_MASK)
                .commit()
    }

    override fun onRecipeCategorySelect(item: RecipeCategory?) {
        Log.d(TAG, item?.name)
    }

    override fun onRecipeClicked(uri: Uri) {
        val intent = Intent(this@MainActivity, RecipeActivity::class.java)
        startActivity(intent)
    }

    companion object {
        private val TAG = "MainActivity"
        var mUser: UserModel.User? = null
    }
}
